### PR TITLE
Mender helm v6.0.0

### PR DESCRIPTION
**Pull request by alfrunes**
I tried to limit the changes to preserve as much backward compatibility as possible. However, this change cannot be done without some breaking changes (especially for workflows-worker).

### CODE REVIEW

File: mender/templates/gui/deployment.yaml [hunk 1]
```diff
@@ -88,9 +89,9 @@ spec:
         - name: HAVE_MULTITENANT
           value: {{ .Values.global.enterprise | quote }}
         - name: MENDER_VERSION
-          value: {{ trimPrefix "mender-" .Values.global.image.tag | quote }}
+          value: {{ trimPrefix "v" .Chart.AppVersion | quote }}
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply mender/templates/gui/deployment.yaml:92]: Hmm... I believe we need to think of something here. I believe this is used for resolving the client version, so maybe we should start using the actual client version here instead :thinking:

[the remark this answers is not in the document]
Comment (mzedel) [reply mender/templates/gui/deployment.yaml:92]: it looks like on hosted this is no longer required as we can rely on the `versions.json` for the downloads section and on prem will be pointed to the install script... I'll create a task to clean up the env vars in the gui image 👌

**Comment by alfrunes** [comment]
After testing it for open-source, I was able to find several issues which I've for now kept in separate commit. I will `git rebase --autosquash ...` before I proceed merging.

**Review by oldgiova — APPROVED** [review]
LGTM :+1: 

Out of scope, but since we're in a breaking release, could we also switch `api_gateway.env.SSL=false` ? Otherwise it won't work. See: https://northerntech.atlassian.net/browse/MEN-7260

**Comment by alfrunes** [comment]
> LGTM 👍 Of course, I cannot test it properly because the Mender v4.0.0 has not been published yet.
> 
> Out of scope, but since we're in a breaking release, could we also switch `api_gateway.env.SSL=false` ? Otherwise it won't work. See: https://northerntech.atlassian.net/browse/MEN-7260

Thanks for reminding me, I forgot about this one. I cherry-picked it from #170 and amended the changelogs.

File: mender/values.yaml [hunk 2]
```diff
@@ -5,12 +5,6 @@ global:
   hosted: false
```
[the remark this answers is not in the document]
Comment (alfrunes) [reply mender/values.yaml:5]: Done

**Review by oldgiova — APPROVED** [review]
Still LGTM

**Pull request merged**